src/load_inputs/load_generators_data.jl

```diff
@@ -185,44 +185,49 @@ function load_generators_data!(setup::Dict, path::AbstractString, inputs_gen::Di
-		start_fuel = convert(Array{Float64}, gen_in[!,:Start_Fuel_MMBTU_per_MW])
-		# Fixed cost per start-up ($ per MW per start) if unit commitment is modelled
+	if setup["UCommit"] >= 1
 		start_cost = convert(Array{Float64}, gen_in[!,:Start_Cost_per_MW])
 		inputs_gen["C_Start"] = zeros(Float64, G, inputs_gen["T"])
-		gen_in[!,:CO2_per_Start] = zeros(Float64, G)
 	end
 
-	# Heat rate of all resources (million BTUs/MWh)
-	heat_rate = convert(Array{Float64}, gen_in[!,:Heat_Rate_MMBTU_per_MWh])
-	# Fuel used by each resource
-	fuel_type = gen_in[!,:Fuel]
-	# Maximum fuel cost in $ per MWh and CO2 emissions in tons per MWh
-	inputs_gen["C_Fuel_per_MWh"] = zeros(Float64, G, inputs_gen["T"])
-	gen_in[!,:CO2_per_MWh] = zeros(Float64, G)
+	# The fuel (including start up fuel) costs and CO2 emissions will be separately tracked in Fuel.jl and CO2.jl
+	# No need to determine C_Fuel_per_MWh and CO2_per_MWh, and C_Start
+	# Only scale the start up cost here
 	for g in 1:G
-		# NOTE: When Setup[ParameterScale] =1, fuel costs are scaled in fuels_data.csv, so no if condition needed to scale C_Fuel_per_MWh
-		inputs_gen["C_Fuel_per_MWh"][g,:] = fuel_costs[fuel_type[g]].*heat_rate[g]
-		gen_in[g,:CO2_per_MWh] = fuel_CO2[fuel_type[g]]*heat_rate[g]
-		gen_in[g,:CO2_per_MWh] *= scale_factor
-		# kton/MMBTU * MMBTU/MWh = kton/MWh, to get kton/GWh, we need to mutiply 1000
 		if g in inputs_gen["COMMIT"]
 			# Start-up cost is sum of fixed cost per start plus cost of fuel consumed on startup.
-			# CO2 from fuel consumption during startup also calculated
-
-			inputs_gen["C_Start"][g,:] = gen_in[g,:Cap_Size] * (fuel_costs[fuel_type[g]] .* start_fuel[g] .+ start_cost[g])
-			# No need to re-scale C_Start since Cap_size, fuel_costs and start_cost are scaled When Setup[ParameterScale] =1 - Dharik
-			gen_in[g,:CO2_per_Start]  = gen_in[g,:Cap_Size]*(fuel_CO2[fuel_type[g]]*start_fuel[g])
-			gen_in[g,:CO2_per_Start] *= scale_factor
-			# Setup[ParameterScale] =1, gen_in[g,:Cap_Size] is GW, fuel_CO2[fuel_type[g]] is ktons/MMBTU, start_fuel is MMBTU/MW,
-			#   thus the overall is MTons/GW, and thus gen_in[g,:CO2_per_Start] is Mton, to get kton, change we need to multiply 1000
-			# Setup[ParameterScale] =0, gen_in[g,:Cap_Size] is MW, fuel_CO2[fuel_type[g]] is tons/MMBTU, start_fuel is MMBTU/MW,
-			#   thus the overall is MTons/GW, and thus gen_in[g,:CO2_per_Start] is ton
+			inputs_gen["C_Start"][g,:] .= gen_in[g,:Cap_Size] * ( start_cost[g])
 		end
 	end
 
 	load_vre_stor_data!(inputs_gen, setup, path)
+
+	
+    # write zeros if col names are not in the gen_in dataframe
+	missing_cols = ["Incremental_Heat_Rate_Segment2", "Incremental_Heat_Rate_Segment3", "Intercept_Fuel_Consumption_Segment1","Intercept_Fuel_Consumption_Segment2", "Intercept_Fuel_Consumption_Segment3",
+	"Biomass", "CO2_Capture_Rate", "CO2_Capture_Cost_per_Metric_Ton"]
+	write_zeros_if_not_exist!(gen_in, missing_cols)
```

Comment:
As you suggested, I will make the number of segments flexible instead of fixed at three. So we don't need to make Incremental_Heat_Rate_Segment and Intercept_Fuel_Consumption_Segment to be zero if these cols do not exist...

But I will keep ["Biomass", "CO2_Capture_Rate", "CO2_Capture_Cost_per_Metric_Ton"] here.. If these cols are not provided then I will make them zeros, otherwise it will cause errors in co2.jl..

Reply:
I don't think this a bad place to put it, and I do like the notion of an `insert zeros` function.

Here's my comments:

* use the existing `ensure_column!` function that I wrote in `load_fuels_data.jl` rather than writing a very similar function. (Maybe we should move that function to `load_dataframe.jl` or somewhere else)

* you can even 'pluralize it' by writing 

```
ensure_column!(df::DataFrame, cols::Vector{AbstractString}, fill_element)
    for col in cols
        ensure_column!(df, col, fill_element)
    end
end
```

* Please rename `missing_cols` to `columns_required_for_co2_module` or something more descriptive.
* Place requires columns on successive lines rather than all on one or two lines. This is good practice since `git` works per-line.

Reply:
ok will use ensure_column to insert zeros!